### DISCUSSION TITLE
I am not able to understand the issue in zustand  and tanstack/react-query

### DISCUSSION THREAD

**Discussion by jaykeerti** [pmndrs/zustand#2256] [General] [answered]
I have a similar situation in my project, which is way larger than the example here.

I am setting up the store in the parent component as soon as the data is loaded, accessing the store in the child component, and using that data to fetch something else using the react query.

When there is a delay in useEffect in the parent component while setting up the store, the child is already rendered and the old state is fetched, which is causing a huge issue.

What is the better approach to doing this?

[https://codesandbox.io/p/sandbox/cocky-thunder-h5frkz?file=%2Fsrc%2FParent.js](url)

Is the above pattern that I am using the correct way to approach it?

Issue:
The issue that I see in my original project is that when I open the child component, the state is not updated; it reflects the older state. Does that mean I need to reset it somehow?

**Comment by dbritto-dev** [comment, marked as answer]
@jaykeerti here you go -> https://codesandbox.io/p/sandbox/cocky-thunder-forked-wxwd76?file=%2Fsrc%2FuserStore.js%3A18%2C10

Some things you need to know:
- The new way to import `create` from zustand is `import { create } from 'zustand'` since default export is deprecated and would be removed in the future
- if you want to store the result from `react-query` on zustand you don't need to use `useQuery()`, you can use `queryClient.fetchQuery()` and then store the value directly into your store
- you should use `useUserStore()` hook instead of `useUserStore.getState()` on react components

**Reply by jaykeerti** [reply]
Thanks for the detailed explaination
